### PR TITLE
fix: Don't AutoScroll to embed when multiple-duration event's booking page is embedded

### LINKED ISSUE #10590: [CAL-2318] embed: remove autofocus inside embed
https://twitter.com/judecornish/status/1687411810570240000

<sub>[CAL-2318](https://linear.app/calcom/issue/CAL-2318/embed-remove-autofocus-inside-embed)</sub>

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

Fixes #10590
The issue occurs only for an event with multiple durations enabled

- Added an e2e to verify it is not present
- Added an eslint to by default error on use of scrollIntoView unless acknowledged

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?
1. Create an event with multiple durations enabled
2. Embed that event's booking page.
3. Notice that the scroll automatically occurs to that embed.

### CODE REVIEW

File: apps/web/playwright/fixtures/users.ts [hunk 1]
```diff
@@ -274,6 +274,12 @@ export const createUsersFixture = (
         { title: "Paid", slug: "paid", length: 30, price: 1000 },
         { title: "Opt in", slug: "opt-in", requiresConfirmation: true, length: 30 },
         { title: "Seated", slug: "seated", seatsPerTimeSlot: 2, length: 30 },
+        {
+          title: "Multiple duration",
```
Comment (hariombalhara) [inline apps/web/playwright/fixtures/users.ts:278]: Allow us to test multiple duration embeds.

File: packages/embeds/embed-core/playwright/tests/inline.e2e.ts [hunk 2]
```diff
@@ -6,29 +6,23 @@ import {
...
+    const embedIframe = await ensureEmbedIframe({ calNamespace, page, pathname: "/pro" });
+    expect(embedIframe).toBeEmbedCalLink(calNamespace, embeds.getActionFiredDetails, {
       searchParams: {
         theme: "dark",
       },
     });
-    // expect(await page.screenshot()).toMatchSnapshot("event-types-list.png");
-    if (!embedIframe) {
-      throw new Error("Embed iframe not found");
-    }
```
Comment (hariombalhara) [inline packages/embeds/embed-core/playwright/tests/inline.e2e.ts:31]: Simplified the test.

File: playwright.config.ts [hunk 3]
```diff
@@ -198,12 +198,14 @@ expect.extend({
     const u = new URL(iframe.url());
 
     const pathname = u.pathname;
-    const expectedPathname = `${expectedUrlDetails.pathname}/embed`;
-    if (expectedPathname && expectedPathname !== pathname) {
-      return {
-        pass: false,
-        message: () => `Expected pathname to be ${expectedPathname} but got ${pathname}`,
-      };
+    if (expectedUrlDetails.pathname) {
```
Comment (hariombalhara) [inline playwright.config.ts:201]: Supports checking pathname only if pathname is expected.

File: packages/embeds/embed-core/playwright/tests/inline.e2e.ts [hunk 4]
```diff
@@ -6,29 +6,23 @@ import {
...
+    embeds,
   }) => {
     await deleteAllBookingsByEmail("embed-user@example.com");
-    await addEmbedListeners("");
-    await page.goto("/?only=ns:default");
+    await embeds.gotoPlayground({ calNamespace: "", url: "/?only=ns:default" });
     const calNamespace = "";
-    const embedIframe = await getEmbedIframe({ calNamespace, page, pathname: "/pro" });
-    expect(embedIframe).toBeEmbedCalLink(calNamespace, getActionFiredDetails, {
-      pathname: "/pro",
```
Comment (hariombalhara) [inline packages/embeds/embed-core/playwright/tests/inline.e2e.ts:23]: No need to check for pathname again. We are getting the embed iframe by pathname only.

**Review by hariombalhara — COMMENTED** [review]
Self review done

File: packages/embeds/embed-core/playwright/tests/inline.e2e.ts [hunk 5]
```diff
@@ -56,6 +50,17 @@ test.describe("Inline Iframe", () => {
...
 
+  test("Ensure iframe doesn't hijack scroll in embed mode", async ({ page, embeds, users }) => {
+    const user = await users.create();
+    const calNamespace = "autoScrollTest";
+    await embeds.gotoPlayground({ calNamespace, url: `?only=ns:autoScrollTest` });
+    const calLink = `${user.username}/multiple-duration`;
+    await page.goto(`/?only=ns:autoScrollTest&cal-link=${calLink}`);
+    const embedIframe = await ensureEmbedIframe({ calNamespace, page, pathname: `/${calLink}` });
+    const finalScrollPosition = await page.evaluate(() => window.scrollY);
+    expect(finalScrollPosition).toBe(0);
```
Comment (sean-brydon) [inline packages/embeds/embed-core/playwright/tests/inline.e2e.ts:61]: Smart :man_cook:

File: packages/eslint-plugin/src/rules/no-scroll-into-view-embed.ts
Comment (sean-brydon) [inline packages/eslint-plugin/src/rules/no-scroll-into-view-embed.ts:1]: Love this idea - as we talked in DMS we should abstract `scrollIntoView` into a custom hook `useScrollIntoView` that includes the check for `isEmbed` Then we can modify this scrollIntoView eslint rule to ensure all `scrollIntoView` usage comes from the hook

**Review by sean-brydon — DISMISSED** [review]
Happy with this - lets address the hook in a follow up as its deffo not blocking. Tested in embed and it works as expected

**Review by alishaz-polymath — APPROVED** [review]

**Pull request merged**